components/server/src/code-sync/code-sync-service.ts

```diff
@@ -303,4 +290,24 @@ export class CodeSyncService {
         return router;
     }
 
+    protected async deleteObjects(userId: string, resourceKey: ServerResource, revs: string[]) {
+        for (let rev of revs) {
+            const obj = toObjectName(resourceKey, rev)
+            const request = new DeleteRequest();
+            request.setOwnerId(userId);
+            request.setExact(obj);
+            const blobsClient = this.blobsProvider.getDefault();
+            try {
+                await util.promisify<DeleteRequest, DeleteResponse>(blobsClient.delete.bind(blobsClient))(request);
+                await this.db.deleteResource(userId, resourceKey, rev);
```

Comment:
I wonder whether it will be better to perform deletion of blob in the transaction of deleting a resource, i.e.:
- open transaction
- delete record from DB
- delete blob from storage (ignore not found exception)
- close transaction

Reply:
I think it's the same with steps now

Reply:
Is not it that if deleteResource fails then we have a record without blob?

Reply:
this one does not seem to be addressed, deletion of blob happens outside of delete DB transaction, that can deleted blob, but no a DB record